packages/compass-global-writes/src/store/index.spec.ts

```diff
@@ -36,108 +74,139 @@ describe('GlobalWritesStore Store', function () {
-        const store = createStore({
-          authenticatedFetch: () =>
-            createJsonResponse({
-              geoSharding: { customZoneMapping: {}, managedNamespaces: [] },
-            }),
-        });
-
-        const promise = store.dispatch(createShardKey(shardKeyData));
-        expect(store.getState().status).to.equal('SUBMITTING_FOR_SHARDING');
-
-        await promise;
-        expect(store.getState().status).to.equal('SHARDING');
+    it('not managed -> failed sharding attempt', async function () {
+      const store = createStore({
+        authenticatedFetch: (uri: string) => {
+          if (uri.includes('/geoSharding')) {
+            return Promise.reject(new Error('Failed to shard'));
+          }
+
+          return createAuthFetchResponse(clusterDetails);
+        },
       });
+      await store.dispatch(fetchClusterShardingData());
+      expect(store.getState().status).to.equal('UNSHARDED');
+      expect(store.getState().managedNamespace).to.equal(undefined);
+
+      const promise = store.dispatch(createShardKey(shardKeyData));
+      expect(store.getState().status).to.equal('SUBMITTING_FOR_SHARDING');
+      await promise;
+      expect(store.getState().status).to.equal('UNSHARDED');
+    });
 
-      it('sets SUBMITTING_FOR_SHARDING state when starting to create shard key and sets to UNSHARDED on failure', async function () {
-        const store = createStore({
-          authenticatedFetch: () => Promise.reject(new Error('error')),
-        });
+    it('when the namespace is managed', async function () {
+      const store = createStore({
+        authenticatedFetch: (uri: string) => {
+          if (uri.includes('/clusters/')) {
+            return createAuthFetchResponse({
+              ...clusterDetails,
+              geoSharding: {
+                ...clusterDetails.geoSharding,
+                managedNamespaces: [managedNamespace],
+              },
+            });
+          }
+
+          if (uri.includes('/deploymentStatus/')) {
+            return createAuthFetchResponse({
+              automationStatus: {
+                processes: [],
+              },
+            });
+          }
+
+          return createAuthFetchResponse({});
+        },
+        automationAgentRequest: (_meta: unknown, type: string) => ({
+          _id: '123',
+          requestType: type,
+        }),
+        automationAgentAwait: (_meta: unknown, type: string) => {
+          if (type === 'getShardKey') {
+            return {
+              response: [
+                {
+                  key: {
+                    location: 'HASHED',
+                    secondary: 'HASHED',
+                  },
+                  unique: false,
+                },
+              ],
+            };
+          }
+        },
+      });
+      await store.dispatch(fetchClusterShardingData());
+      // expect(store.getState().status).to.equal('SHARD_KEY_CORRECT'); //  no idea why this does not work
```

Comment:
Is the state not matching or the check? What state do you see here? (we probably want to either remove this comment altogether or resolve, a bit weird to just leave this comment 🙂)

Reply:
I was seeing NOT_READY. Putting the check behind waitFor helped, but I'm not sure why it was needed 🤔

Reply:
I guess it's because this fetch action is not the one that actually will update the state, and the one that will is not awaited inside this one:

![image](https://github.com/user-attachments/assets/dec32e48-649f-4099-a43a-9992b75c3ef5)

Not sure if it's this intentional or not

Reply:
ah right! I was looking at this code and read it wrong 🤦  outside of tests it currently doesn't make difference if we wait for this one, since the `fetchClusterShardingData` is not awaited.